src/components/Accordion/Accordion.tsx

```diff
@@ -82,6 +86,12 @@ class Accordion extends AutoControlledComponent<any, any> {
 
   state: any = { activeIndex: [0] }
 
+  constructor(props, state) {
+    super(props, state)
+    const a11yType: string = props.a11yType
+    this.accBehavior = A11yBehaviorFactory.createBehavior(A11yBehaviorType[a11yType])
```

Comment:
Maybe we should add logic in the componentWillReceiveProps() in order to check if the prop.a11yType is changed to accordingly reinitialize the accBehavior, because with the current implementation, it is initialized only in the constructor. (I haven't seen the exact implementation, but you can test this with changing the prop a11yType in run time). In addition, if this is used in all components, maybe we can extract this logic in the UIComponent.

Reply:
I think our goal is to set accessibility behavior only once, so the component wouldn't change during runtime. Let's assume we have a list which behaves as a selectable list with all needed accessibility roles. So, we don't want this behavior to change somehow during runtime.

Reply:
If that's the case, then it is ok to have this in the constructor, maybe we should just consider making this generic and not adding it in all components.

Reply:
I would like to know how we can make it generic. The only thing that each component should have set default accessibility behavior. Do have ideas how to improve it?